Fix go version in lint job

ricky-rav opened the pull request:
Add golang installation step to lint job, since `golangci-lint` might install a later version by design: https://github.com/golangci/golangci-lint-action/issues/75

Signed-off-by: Riccardo Ravaioli <rravaiol@redhat.com>

dcbw commented:
/lgtm

girishmg commented:
@ricky-rav @dcbw the lint errors are with Go 1.19. With 1.18 it's fine. Not sure why the Github workflow sometimes install GO 1.19 instead of 1.18.4.

girishmg commented:
@ricky-rav  for example check this run here: https://github.com/ovn-org/ovn-kubernetes/runs/7746638765?check_suite_focus=true#step:3:50 ... 

```
GOPATH="/home/runner/go"
  GOPRIVATE=""
  GOPROXY="https://proxy.golang.org/,direct"
  GOROOT="/opt/hostedtoolcache/go/1.19.0/x64"
  GOSUMDB="sum.golang.org"
  GOTMPDIR=""
  GOTOOLDIR="/opt/hostedtoolcache/go/1.19.0/x64/pkg/tool/linux_amd64"
  GOVCS=""
```

However, in the workflow the Go version we need is 1.18.4 .. https://github.com/ovn-org/ovn-kubernetes/blob/master/.github/workflows/test.yml#L10

ricky-rav commented:
Thanks for pointing that out, @girishmg ! I have to give it a closer look, then...

ricky-rav commented:
OK, so I added an extra step in the lint job in order to install explicitly the version of go that we want and we call now `golangci-lint` with `--skip-golang-installation`, so that it won't go and fetch a different go version. From the logs:

```
2022-08-09T16:31:44.8468080Z ##[group]Run golangci/golangci-lint-action@v2
2022-08-09T16:31:44.8469036Z with:
2022-08-09T16:31:44.8469245Z   version: v1.46
2022-08-09T16:31:44.8469517Z   working-directory: go-controller
2022-08-09T16:31:44.8469874Z   args: --modules-download-mode=vendor --timeout=15m0s --verbose
2022-08-09T16:31:44.8470199Z   skip-go-installation: true
[...]
2022-08-09T16:31:44.9969597Z ##[group]prepare environment
2022-08-09T16:31:44.9994932Z Finding needed golangci-lint version...
2022-08-09T16:31:45.0164586Z Skipping the installation of Go
2022-08-09T16:31:45.0422762Z Requested golangci-lint 'v1.46', using 'v1.46.2', calculation took 43ms
2022-08-09T16:31:45.0445069Z Installing golangci-lint v1.46.2...
```

I removed the previous changes where I was replacing `ioutils` with `os` calls, since that turned out not to be the problem.

@girishmg @dcbw PTAL

Thanks!
